pkg/api/api.go

```diff
@@ -63,6 +63,8 @@ func prefixError(name string, reqs []Request) []Request {
 		rq := r
 
 		res = append(res, func(ctx context.Context, ch chan<- *Record) error {
+			logger.Info(fmt.Sprintf("Snapshot: starting %s", name))
```

Comment:
Don't use a global variable. Your options are very limited here, I suggest trying to pass in the logger through the context using `FromContextOrDiscard).

I'm not sure there's enough value in logging every action every time to justify using `Info`. I think it's useful to print this info at higher verbosity (e.g. level 4 or above).

`logr.Logger` is a structured logger. You should give it key value pairs so the information can be more easily extracte by logging tools such as Observe.

Putting things all together, something like this would probably work:

```
func(ctx context.Context, ch chan<- *Record) error {
  logger := logr.FromContextOrDiscard(ctx)
  logger.V(6).Log("snapshot start", "action", name)
  defer logger.V(6).Log("snapshot complete", "action", name)
```